src/controllers/expenses.controller.js

```diff
@@ -0,0 +1,86 @@
+const { expensesService } = require('../services/expense.services.js');
+const { usersService } = require('../services/user.services.js');
+
+const getExpenses = (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  const categoryList = categories ? categories.split(',') : [];
+
+  const expenses = expensesService.getExpenses({
+    userId,
+    categoryList,
+    from,
+    to,
+  });
+
+  res.json(expenses);
+};
+
+const createExpense = (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+
+  if (!userId || !spentAt || !title || !amount || !category || !note) {
+    return res.sendStatus(400);
+  }
+
+  const user = usersService.getUserById(userId);
+
+  if (!user) {
+    return res.sendStatus(400);
+  }
+
+  const expense = expensesService.createExpense({
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  });
+
+  res.status(201).json(expense);
+};
+
+const getExpenseById = (req, res) => {
+  const expense = expensesService.getExpenseById(+req.params.id);
+
+  if (!expense) {
+    return res.sendStatus(404);
```

Comment:
The error response for a non-existent expense should include a descriptive message. Consider using `res.status(404).json({ message: 'Expense not found' });` to provide more context.